pkg/shares/share_splitting.go

```diff
@@ -72,21 +72,22 @@ func SplitTxs(txs coretypes.Txs) (txShares []Share, pfbShares []Share, shareRang
 	return txShares, pfbShares, mergeMaps(txMap, pfbMap), nil
 }
 
-func SplitBlobs(cursor int, indexes []uint32, blobs []coretypes.Blob, useShareIndexes bool) ([]Share, error) {
-	if useShareIndexes && len(indexes) != len(blobs) {
-		return nil, ErrIncorrectNumberOfIndexes
+// SplitBlob splits the provided blob into shares.
+func SplitBlob(blob coretypes.Blob) ([]Share, error) {
+	writer := NewSparseShareSplitter()
+	if err := writer.Write(blob); err != nil {
+		return nil, err
 	}
+	return writer.Export(), nil
+}
+
+// SplitBlobs splits the provided blobs into shares.
+func SplitBlobs(blobs []coretypes.Blob) ([]Share, error) {
```

Comment:
What about unifying them into a single `SplitBlobs` with variadic param? 
Looks like a perfect match.

Reply:
Can do! Thanks for the suggestion.

Reply:
Resolved by [10f5f55](https://github.com/celestiaorg/celestia-app/pull/1895/commits/10f5f552855e7cb8888776fb6a34b3fbab25513d)